### PR TITLE
Drop IsDeprecatedWeakRefSmartPointerException from RemoteLegacyCDMSession

### PR DESCRIPTION
**Pull request by rniwa**
#### 103c111a8e18da3904ecc4b45941814c8a43ebe5
<pre>
Drop IsDeprecatedWeakRefSmartPointerException from RemoteLegacyCDMSession
<a href="https://bugs.webkit.org/show_bug.cgi?id=281769">https://bugs.webkit.org/show_bug.cgi?id=281769</a>

Reviewed by NOBODY (OOPS!).

This PR makes WebCore::LegacyCDMSession ref counted in order to remove
IsDeprecatedWeakRefSmartPointerException from WebKit::RemoteLegacyCDMSession.

* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDM.cpp:
(WebCore::LegacyCDM::createSession):
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDM.h:
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDMPrivate.h:
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDMPrivateClearKey.cpp:
(WebCore::LegacyCDMPrivateClearKey::createSession):
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDMPrivateClearKey.h:
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDMPrivateMediaPlayer.cpp:
(WebCore::CDMPrivateMediaPlayer::createSession):
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDMPrivateMediaPlayer.h:
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDMSessionClearKey.cpp:
(WebCore::LegacyCDMSessionClearKey::create):
(WebCore::LegacyCDMSessionClearKey::LegacyCDMSessionClearKey):
(WebCore::LegacyCDMSessionClearKey::generateKeyRequest):
(WebCore::LegacyCDMSessionClearKey::releaseKeys):
(WebCore::LegacyCDMSessionClearKey::update):
(WebCore::LegacyCDMSessionClearKey::cachedKeyForKeyID const):
(WebCore::CDMSessionClearKey::CDMSessionClearKey): Deleted.
(WebCore::CDMSessionClearKey::generateKeyRequest): Deleted.
(WebCore::CDMSessionClearKey::releaseKeys): Deleted.
(WebCore::CDMSessionClearKey::update): Deleted.
(WebCore::CDMSessionClearKey::cachedKeyForKeyID const): Deleted.
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDMSessionClearKey.h:
* Source/WebCore/Modules/encryptedmedia/legacy/WebKitMediaKeySession.h:
* Source/WebCore/platform/graphics/LegacyCDMSession.h:
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::MediaPlayer::createSession):
* Source/WebCore/platform/graphics/MediaPlayer.h:
* Source/WebCore/platform/graphics/MediaPlayerPrivate.h:
(WebCore::MediaPlayerPrivateInterface::createSession):
* Source/WebCore/platform/graphics/avfoundation/CDMPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/CDMPrivateMediaSourceAVFObjC.mm:
(WebCore::CDMPrivateMediaSourceAVFObjC::createSession):
* Source/WebCore/platform/graphics/avfoundation/cf/CDMSessionAVFoundationCF.cpp: Removed.
* Source/WebCore/platform/graphics/avfoundation/cf/CDMSessionAVFoundationCF.h: Removed.
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.h:
(WebCore::CDMSessionAVContentKeySession::hasContentKeySession const): Deleted.
(WebCore::CDMSessionAVContentKeySession::logClassName const): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.mm:
(WebCore::CDMSessionAVContentKeySession::create):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVFoundationObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVFoundationObjC.mm:
(WebCore::CDMSessionAVFoundationObjC::create):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::createSession):
* Source/WebCore/testing/LegacyMockCDM.cpp:
(WebCore::LegacyMockCDM::createSession):
(WebCore::MockCDMSession::create):
* Source/WebCore/testing/LegacyMockCDM.h:
* Source/WebKit/GPUProcess/media/RemoteLegacyCDMSessionProxy.h:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::createSession):
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDM.cpp:
(WebKit::RemoteLegacyCDM::createSession):
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDM.h:
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDMSession.cpp:
(WebKit::RemoteLegacyCDMSession::create):
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDMSession.h:
</pre>
https://github.com/WebKit/WebKit/commit/103c111a8e18da3904ecc4b45941814c8a43ebe5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/72742 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52167 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/25541 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/76933 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/23976 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/74857 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59972 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/23836 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/57192 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15687 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/75809 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/47170 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62606 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/37620 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43817 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/20072 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/22305 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65671 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20431 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/78608 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16988 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/19609 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/65641 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/17036 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62614 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64918 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/13226 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6875 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47965 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/2752 "") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/49032 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/50327 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/48777 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/platform/graphics/avfoundation/CDMPrivateMediaSourceAVFObjC.mm [hunk 1]
```diff
@@ -125,17 +125,17 @@ static bool queryDecoderAvailability()
...
+RefPtr<LegacyCDMSession> CDMPrivateMediaSourceAVFObjC::createSession(LegacyCDMSessionClient& client)
 {
     String keySystem = m_cdm->keySystem(); // Local copy for StringView usage
     auto parameters = parseKeySystem(m_cdm->keySystem());
     ASSERT(parameters);
     if (!parameters)
         return nullptr;
 
-    auto session = makeUnique<CDMSessionAVContentKeySession>(WTFMove(parameters.value().protocols), parameters.value().version, *this, client);
+    auto session = CDMSessionAVContentKeySession::create(WTFMove(parameters.value().protocols), parameters.value().version, *this, client);
```
Comment (cdumez) [inline Source/WebCore/platform/graphics/avfoundation/CDMPrivateMediaSourceAVFObjC.mm]: nit: `auto` -> `Ref`

**Review by cdumez — APPROVED** [review]
LGTM

File: Source/WebCore/platform/graphics/avfoundation/CDMPrivateMediaSourceAVFObjC.mm (hunk 1 above)
Comment (rniwa) [reply Source/WebCore/platform/graphics/avfoundation/CDMPrivateMediaSourceAVFObjC.mm]: Fixed.

**Comment by rniwa** [comment]
This PR is obsolete now.

**Pull request closed without merging**